src/main/java/io/camunda/zeebe/bpmnassert/testengine/RecordStreamSourceImpl.java

```diff
@@ -122,4 +133,45 @@ public void print(final boolean compact) {
       System.out.println("---------------------------");
     }
   }
+
+  private void updateWithNewRecords() {
+    if (lastPosition < 0) {
+      logStreamReader.seekToFirstEvent();
+    } else {
+      logStreamReader.seekToNextEvent(lastPosition);
```

Comment:
:question: Does this work as expected when there is no event after last position?

Reply:
It won't throw any errors or that sort of thing. It returns `false` if there are no events after this position. This means `logStreamReader.hasNext()` resolves to `false` as well and the list of records doesn't get updated as there are no new records.